library/Fission/Types.hs

```diff
@@ -563,3 +567,30 @@ instance Domain.Creator Fission where
   create domainName userId zoneId now =
     runDB $ Domain.create domainName userId zoneId now
 
+instance Challenge.Creator Fission where
+  create email = 
+    runDB $ Challenge.create email
+
+instance Challenge.Verifier Fission where
+  verify challenge = 
+    runDB $ Challenge.verify challenge
+
+instance MonadEmail Fission where
+  sendVerifyEmail recipient@(Email.Recipient { name }) (Challenge hash) = do
+    httpManager      <- asks tlsManager
+    Host baseHostUrl <- asks host
+    Host sibUrl      <- asks sibUrl
+    apiKey           <- asks sibApiKey
+    templateId       <- asks sibTemplateId
+
+    let
+      env = mkClientEnv httpManager sibUrl
+      path = "user/verify_email/" <> Text.unpack hash
```

Comment:
is there a better option instead of hard coding this? :thinking: I'm not seeing any methods in Servant to go from `API -> Text`

Reply:
I haven't tried to do this, but looking at Stackage: https://www.stackage.org/haddock/lts-16.3/servant-0.16.2/Servant-API.html#v:safeLink

Reply:
You can 100% use Servant for templating, so they for sure have some way of generating links. I'd assume relative links are an option.

Reply:
Also potentially helpful: https://docs.servant.dev/en/stable/cookbook/generic/Generic.html?highlight=link#links

Though we're not using `servant-generic` at the moment. Not opposed to it, though. It's known to be a bit on the slow side, but in this case we'll _probably_ generate it once it'll stay memoized for future use.

Reply:
Here's exactly the aformentioned relative link function:

https://www.stackage.org/haddock/lts-16.3/servant-lucid-0.9/Lucid-Servant.html#v:safeRelHref_

Lucid is one of the templating libraries. It's very fast, typesafe, and "good". If/when we want to serve HTML from the API, we'd probably use either Lucid or Blaze.

We don't lose anything by including Lucid as a dependency other than adding a bit of time downloading the package the first time. If this is the shortest path, do that 👍

Reply:
# ASIDE

While I'm looking at that those two:

https://mmhaskell.com/blog/2020/3/16/lucid-another-html-option

I've used Blaze before (years ago) and liked it. Lucid is newer. I vaguely remember comparing them around the New Year, and feeling like Lucid was probbaly the better choice, but they're _very_ similar.

Reply:
sweet got it workin :man_dancing: